src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -1085,22 +1086,27 @@ private void filterResultFromSnowpipeScan(
 
     private void purge(List<String> files) {
       if (!files.isEmpty()) {
-        LOGGER.debug(
-            "Purging loaded files for pipe:{}, loadedFileCount:{}, loadedFiles:{}",
-            pipeName,
-            files.size(),
-            Arrays.toString(files.toArray()));
+        String logString = String.format(
+            "Purging loaded files for pipe: %s, loadedFileCount: %d", pipeName, files.size()
+        );
+        LOGGER.info(logString + prepareFilesOffsetsLogString(
+                files, "loadedFiles", LOGGER.isDebugOrTraceEnabled()
+            )
+        );
         conn.purgeStage(stageName, files);
       }
     }
 
     private void moveToTableStage(List<String> failedFiles) {
       if (!failedFiles.isEmpty()) {
-        LOGGER.debug(
-            "Moving failed files for pipe:{} to tableStage failedFileCount:{}, failedFiles:{}",
-            pipeName,
-            failedFiles.size(),
-            Arrays.toString(failedFiles.toArray()));
+        String logString = String.format(
```

Comment:
It will be easier to use String extrapolation in the LOGGER itself instead of calling format:
```
LOGGER.info("Moving failed files for pipe: {} to tableStage failedFileCount: {}", pipeName, files.size())
```
It's also better in terms of performance and security.